app/models/signon_identity.rb

```diff
@@ -15,63 +17,73 @@ def from_omniauth(omniauth_auth)
     end
 
     def from_session_data(data)
-      user = User.find(data.fetch('user_id'))
       new(
-        user,
+        User.find(data.fetch('user_id')),
         full_name: data.fetch('full_name'),
         profile_url: data.fetch('profile_url'),
-        logout_url: data.fetch('logout_url')
+        logout_url: data.fetch('logout_url'),
+        available_organisations: data.fetch('available_organisations'),
+        current_organisation: data.fetch('current_organisation')
       )
     rescue KeyError
       raise InvalidSessionData
     end
 
   private
 
-    # rubocop:disable Metrics/MethodLength
     def find_or_create_authorized_user(info)
       email = info.fetch('email')
-      permissions = info.fetch('permissions')
-
-      sso_orgs = permissions.map { |p| p.fetch('organisation') }
-      estates = Estate.where(sso_organisation_name: sso_orgs)
+      sso_orgs = valid_pvb_orgs(info)
 
-      unless estates.any?
+      unless sso_orgs.any?
+        permissions = info.fetch('permissions')
         Rails.logger.info \
           "User #{email} has no valid permissions: #{permissions}"
         return nil
       end
 
-      User.find_or_create_by!(email: email) do |user|
-        # TODO: Allow the user to access multiple estates, or to switch estates
-        user.estate = estates.first
-      end
+      User.find_or_create_by!(email: email)
     end
-    # rubocop:enable Metrics/MethodLength
 
     def extract_additional_data(info)
-      first_name = info.fetch('first_name')
-      last_name = info.fetch('last_name')
-      full_name = [first_name, last_name].reject(&:empty?).join(' ')
-
       links = info.fetch('links')
+      sso_orgs = valid_pvb_orgs(info)
 
       {
-        full_name: full_name,
+        full_name: full_name_from_additional_data(info),
         profile_url: links.fetch('profile'),
-        logout_url: links.fetch('logout')
+        logout_url: links.fetch('logout'),
+        available_organisations: sso_orgs,
+        current_organisation: sso_orgs.first
```

Comment:
This could do with a comment explaining that we default to the first organisation.

I'm assuming this is what we want? That if they log out and back in again their default estate is reset?